mne/utils.py

```diff
@@ -1872,17 +1872,23 @@ def _time_mask(times, tmin=None, tmax=None, strict=False, sfreq=None):
     """Helper to safely find sample boundaries"""
     tmin = -np.inf if tmin is None else tmin
     tmax = np.inf if tmax is None else tmax
-    mask = (times >= tmin)
-    mask &= (times <= tmax)
     if not strict:
         if sfreq is not None:
             # Push to nearest sample first
             if np.isfinite(tmin):
                 tmin = round(tmin * sfreq) / sfreq
+            else:
+                tmin = times[0]
             if np.isfinite(tmax):
                 tmax = round(tmax * sfreq) / sfreq
-        mask |= isclose(times, tmin)
-        mask |= isclose(times, tmax)
+            else:
+                tmax = times[-1]
+        deltas = np.abs(times - tmax)  # Find nearest times
+        tmax = times.flat[np.where(deltas == deltas.min())[0]][-1]
```

Comment:
Hmm. How about rounding upwards on a tie? I think it could work, although this is kind of difficult to get my head around the whole thing with floats.